services-api/src/main/java/io/scalecube/services/methods/ServiceMethodInvoker.java

```diff
@@ -106,6 +141,25 @@ public ServiceMethodInvoker(
     return throwable != null ? Mono.error(throwable) : result;
   }
 
+  @SuppressWarnings("unchecked")
+  private Mono<Object> authenticate(ServiceMessage message) {
+    return Mono.defer(
+        () -> {
+          if (!methodInfo.isAuth()) {
+            return Mono.empty();
+          }
+          Object credentials = message.credentials();
+          if (credentials == null) {
```

Comment:
It's redundant check, IMHO we shouldn't check it, we should just call `authenticatorImpl` and let it make a decision.

Reply:
Removed.

Reply:
@segabriel 
Suppose as an authenticator implementor now I have to deal with `null` as `credentials` in my function. Ok. What alternative to throwing Exception do I have?